pkg/hsskey/hsskey.go

```diff
@@ -148,31 +148,40 @@ func getHssFromIpmi(verbose bool, verboseDangerous bool) ([][]uint8, error) {
 //     searching will be in the format: "/sys/bus/i2c/devices/{eepromPattern}/eeprom".
 //     For example, 0-005*
 //     An empty string "" will skip the attempt to read from EEPROM.
-func GetAllHss(verbose bool, verboseDangerous bool, eepromPattern string) ([][]uint8, error) {
+func GetAllHss(verbose bool, verboseDangerous bool, eepromPattern string, hssFiles string) ([][]uint8, error) {
 	// Attempt to get HSS from IPMI.
 	hssList, err := getHssFromIpmi(verbose, verboseDangerous)
-	if err == nil && len(hssList) > 0 {
-		// If it successfully reads from IPMI and the list is not empty, return the result.
-		return hssList, nil
+	if err != nil || len(hssList) == 0 {
+		log.Printf("Failed to get HSS key from IPMI: %v", err)
```

Comment:
Printf from package is generally a bad idea; can you return a wrapped error?

Reply:
Willing to take advice on the go paradigm of returning errors. The existing methodology for HSS retrieval is to throw all retrieval functions here and each are indiscriminately tried. Older use-cases got HSS _only_ from IPMI OR _only_ from eeprom. Newer use-cases will have 1 set of HSS from multiple sources.

If 1 source succeeds and one fails, I believe we want unlocking to continue. Our return structure is pretty strict with return value IFF err==nil. We also want platforms to share a common uinit, so separating tried protocols I don't think is ideal.  

I'd think a print here would be fine as a med-term solution. The boot must continue.

Reply:
Marking resolved. Switched to a io.WriteString from an argument.